### DISCUSSION TITLE
Potential Bug: setValue does not always work for unregistered inputs

### DISCUSSION THREAD

**Discussion by joniaiuser** [react-hook-form/react-hook-form#6104] [RFC]
I have this piece of code in my app

```typescript
const removeFile = useCallback(
		({ _id, actionType }: { _id?: string; actionType: ContractActionType }) => {
			const newAttachmentArray =
				attachments.get(actionType)?.filter((att) => att._id !== _id) ?? [];

			setAttachments(
				produce((prev) => prev.set(actionType, newAttachmentArray))
			);
			const idx = getIndex(actionType);
			const currentData = getValues(
				`loanContractActions.${idx}` as "loanContractActions.0"
			);
			register(`loanContractActions.${idx}` as "loanContractActions.0");
			setValue(
				`loanContractActions.${idx}` as "loanContractActions.0",
				{ ...currentData, attachments: newAttachmentArray },
				{ shouldDirty: true }
			);
		},
		[attachments, setValue, getValues, setAttachments, getIndex, register]
	);
```

If I remove the line 
```register(`loanContractActions.${idx}` as "loanContractActions.0");```
There will be instances where after calling `router.replace(router.asPath)` in nextjs, the `setValue` will be unable to set the value of the `attachments` array back to an empty array. It will work again after a hard refresh.

I have yet to do a repro in code sandbox, as I managed to fix it using `register`. But this is different behaviour from what is suggested in the documentation where register is only 'recommended'. 

**I am using version 7.12.1 react-hook-form and NextJS version 11.0.1**

**Comment by bluebill1049** [comment]
Try with the next version `v7.12.1-next.0` it's always a little bit harder to setValue with out register tho, consider hook form is all about inptus.

**Reply by joniaiuser** [reply]
I see. Just gave it a try. Same issue if I remove the register line. Adding it back fixes it.
